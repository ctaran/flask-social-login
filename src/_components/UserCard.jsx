import React from 'react';
import { Card, Icon, Image } from 'semantic-ui-react';

export default function UserCard(props) {
    return (
        <Card>
            <Image src={require(`../_images/user${props.user.id}.png`)} wrapped ui={false} />
            <Card.Content>
            <Card.Header>{props.user.name}</Card.Header>
            <Card.Meta>
                <span className='date'>{props.user.email}</span>
            </Card.Meta>
            <Card.Description>
                {props.user.name} has the {props.user.role} role
            </Card.Description>
            </Card.Content>
            <Card.Content extra>
            <a>
                <Icon name='user' />
                {props.user.name} is coming from {props.user.origin}
            </a>
            </Card.Content>
        </Card>
    );
}