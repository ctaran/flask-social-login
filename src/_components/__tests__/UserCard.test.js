import { shallow } from "enzyme";
import React from "react";
import { render, unmountComponentAtNode } from "react-dom";
import { act } from "react-dom/test-utils";

import UserCard from "../UserCard";

let container = null;
beforeEach(() => {
  // setup a DOM element as a render target
  container = document.createElement("div");
  document.body.appendChild(container);
});

afterEach(() => {
  // cleanup on exiting
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

let testUser1 = {
    id: 1,
    name: "admin",
    email: "admin@gmail.com",
    role: "ADMIN",
    origin: "INTERNAL"
}

it("should render initial layout", () => {
  const component = shallow(<UserCard user={testUser1}/>);
  expect(component.getElements()).toMatchSnapshot();
});