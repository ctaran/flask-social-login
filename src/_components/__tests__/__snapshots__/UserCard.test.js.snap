// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render initial layout 1`] = `
Array [
  <Card>
    <Image
      as="img"
      src="user1.png"
      ui={false}
      wrapped={true}
    />
    <CardContent>
      <CardHeader>
        admin
      </CardHeader>
      <CardMeta>
        <span
          className="date"
        >
          admin@gmail.com
        </span>
      </CardMeta>
      <CardDescription>
        admin
         has the 
        ADMIN
         role
      </CardDescription>
    </CardContent>
    <CardContent
      extra={true}
    >
      <a>
        <Icon
          as="i"
          name="user"
        />
        admin
         is coming from 
        INTERNAL
      </a>
    </CardContent>
  </Card>,
]
`;
